lib/Frontend/Frontend.cpp

```diff
@@ -261,175 +264,281 @@ ModuleDecl *CompilerInstance::getMainModule() {
+/// Implicitly import the SwiftOnoneSupport module in non-optimized
+/// builds. This allows for use of popular specialized functions
+/// from the standard library, which makes the non-optimized builds
+/// execute much faster.
+static bool shouldImplicityImportSwiftOnoneSupportModule(
+    SILOptions::SILOptMode optimization,
+    FrontendOptions::ActionType requestedAction) {
+  return ((optimization <= SILOptions::SILOptMode::None &&
+           (requestedAction == FrontendOptions::EmitObject ||
+            requestedAction == FrontendOptions::Immediate ||
+            requestedAction == FrontendOptions::EmitSIL)) ||
+          (optimization == SILOptions::SILOptMode::None &&
+           requestedAction >= FrontendOptions::EmitSILGen));
+}
+
+void CompilerInstance::performSema() {
+  SharedTimer timer("performSema");
+  Context->LoadedModules[MainModule->getName()] = getMainModule();
+  
+  if (Invocation.getInputKind() == InputFileKind::IFK_SIL) {
+    assert(BufferIDs.size() == 1);
+    assert(MainBufferID != NO_SUCH_BUFFER);
+    createSILModule();
+  }
+  
+  if (Invocation.getImplicitModuleImportKind() == SourceFile::ImplicitModuleImportKind::Stdlib) {
+    if (!loadStdlibAndMaybeSwiftOnoneSupport())
+    return;
   }
+  
+  ImplicitImports implicitImports(*this);
+  
+  if (Invocation.getInputKind() == InputFileKind::IFK_Swift_REPL) {
+    createREPLFileWithImports(implicitImports);
+    return;
   }
 
+  std::unique_ptr<DelayedParsingCallbacks> DelayedCB(
+      computeDelayedParsingCallback());
+
+  PersistentParserState PersistentState;
+
+  // Make sure the main file is the first file in the module, so do this now.
+  createMainFileWithImportsAndAddToModule(implicitImports);
+
+  bool hadLoadError = parsePartialModulesAndLibraryFiles(implicitImports, PersistentState, DelayedCB.get());
+  if (hadLoadError)
+    return;
+
+  // Parse main file last in order to make sure that it can use decls from other
+  // files in the module.
+  checkTypesWhileParsingMain(PersistentState, DelayedCB.get());
+}
+
+CompilerInstance::ImplicitImports::ImplicitImports(CompilerInstance &compiler) {
+  kind = compiler.Invocation.getImplicitModuleImportKind();
+
   auto clangImporter =
-    static_cast<ClangImporter *>(Context->getClangModuleLoader());
-
-  ModuleDecl *underlying = nullptr;
-  if (options.ImportUnderlyingModule) {
-    underlying = clangImporter->loadModule(SourceLoc(),
-                                           std::make_pair(MainModule->getName(),
-                                                          SourceLoc()));
-    if (!underlying) {
-      Diagnostics.diagnose(SourceLoc(), diag::error_underlying_module_not_found,
-                           MainModule->getName());
-    }
+  static_cast<ClangImporter *>(compiler.Context->getClangModuleLoader());
+  
+  objCModuleUnderlyingMixedFramework =
+  compiler.Invocation.getFrontendOptions().ImportUnderlyingModule
+  ? compiler.importUnderlyingModule(clangImporter)
+  : nullptr;
```

Comment:
Not sure what happened to all the indentation in this method.

Reply:
Should be fixed now.